### DISCUSSION TITLE
does minio expose api request latency metric to prometheus?

### DISCUSSION THREAD

**Discussion by lddlww** [minio/minio#14293] [Q&A] [answered]
i want to use grafana which get metric data from prometheus to monitor minio server request latency , how can i do?

**Comment by harshavardhana** [comment, marked as answer]
Check here https://github.com/minio/minio/blob/master/docs/metrics/prometheus/list.md - what we expose is a more meaningful 

minio_s3_time_ttfb_seconds_distribution

**Reply by lddlww** [reply]
thanks for reply
but when i use prometheus ```histogram_quantile```  to get latency, it return ```NaN```
![image](https://user-images.githubusercontent.com/27006237/153691549-d0b5eee0-acc1-42ec-a58e-e43bb1235c69.png)

**Reply by lddlww** [reply]
reference prometheus document：
```
The histogram_quantile() function interpolates quantile values by assuming a linear distribution within a bucket. The highest bucket must have an upper bound of +Inf. (Otherwise, NaN is returned.)
```
seems ```minio_s3_time_ttfb_seconds_distribution``` lack ```+Inf``` bucket?

**Reply by harshavardhana** [reply]
You need to just use our garafana do not need to re- engineer this ..

**Reply by lddlww** [reply]
ok, thanks

**Reply by WolfspiritM** [reply]
@harshavardhana We're currently trying to figure out a way to get the request duration rate in grafana. I can't see any usage of minio_s3_time_ttfb_seconds_distribution. I can only find the request rate per second but not the duration. Am I missing something here?

**Reply by bem3** [reply]
> You need to just use our garafana do not need to re- engineer this ..

https://grafana.com/grafana/dashboards/13502-minio-dashboard/ You mean this one? But I do not find minio_s3_time_ttfb_seconds_distribution metric. Please take a look at my problem, Thank U! @harshavardhana

**Reply by ShapedTime** [reply]
I am struggling to calculate 0.9 quantile at the moment

**Comment by morningcafe** [comment]
I'm doing this expression for the metric:
```
histogram_quantile(0.9, sum by(le) (minio_bucket_requests_ttfb_seconds_distribution{bucket="100kb-bucket"}))
```
to caculate 90th. 
Can you tell I'm doing right or not @harshavardhana :?
